api/src/main/java/javax/enterprise/inject/spi/container/ContainerBuilder.java

```diff
@@ -0,0 +1,313 @@
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package javax.enterprise.inject.spi.container;
+
+import javax.enterprise.inject.spi.AnnotatedType;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.Extension;
+import java.lang.annotation.Annotation;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+/**
+ * CDI container builder for Java SE.
+ * It is obtained by calling the {@link ContainerBuilder#getContainer()} static method
+ * <p>
+ * <p>
+ * Typical usage looks like this:
+ * </p>
+ * <p>
+ * <pre>
+ * CDISE<Object> cdi = ContainerBuilder.getContainer().initialize();
+ * cdi.select(Foo.class).get();
+ * cdi.event().select(Bar.class).fire(new Bar());
+ * cdi.shutdown();
+ * </pre>
+ * <p>
+ * <p>
+ * The {@link CDISE} interface implements AutoCloseable:
+ * </p>
+ * <p>
+ * <pre>
+ * try (CDISE<Object> cdi = ContainerBuilder.getContainer().initialize()) {
+ *     cdi.select(Foo.class).get();
+ * }
+ * </pre>
+ * <p>
+ * <p>
+ * By default, the discovery is enabled so that all beans from all discovered bean archives are considered. However, it's possible to define a "synthetic" bean
+ * archive, or the set of bean classes and enablement respectively:
+ * </p>
+ * <p>
+ * <pre>
+ * CDISE<Object> cdi = ContainerBuilder.getContainer().beanClasses(Foo.class, Bar.class).alternatives(Bar.class).initialize());
+ * </pre>
+ * <p>
+ * <p>
+ * Moreover, it's also possible to disable the discovery completely so that only the "synthetic" bean archive is considered:
+ * </p>
+ * <p>
+ * <pre>
+ * CDISE<Object> cdi = ContainerBuilder.getContainer().disableDiscovery().beanClasses(Foo.class, Bar.class).initialize());
+ * </pre>
+ * <p>
+ * <p>
+ * <p>
+ * In the same manner, it is possible to explicitly declare interceptors, decorators, extensions and implementation specific options using the builder.
+ * </p>
+ * <p>
+ * <pre>
+ * ContainerBuilder container = ContainerBuilder.getContainer()
+ *    .disableDiscovery()
+ *    .packages(Main.class, Utils.class)
+ *    .interceptors(TransactionalInterceptor.class)
+ *    .property("property", true);
+ * CDISE<Object> cdi = container.initialize();
+ * </pre>
+ *
+ * @since 2.0
+ * @author Antoine Sabot-Durand
+ * @author Martin Kouba
+ * @author John D. Ament
+ */
+public abstract class ContainerBuilder {
```

Comment:
You're right, the final name for CDISE is important there.
Reading the notes/mails we have the following proposals:
* CDIContainer
* CDISE
* SeContainer
* StandaloneContainer
* UserContainer

I like 3 last since they introduce the idea of user managed container or SE. CDISE is ok, I'm not a big fan of its casing but it makes sense...